hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveViewCatalog.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Collections;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.view.ViewCatalogTests;
+import org.assertj.core.api.Assumptions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class TestHiveViewCatalog extends ViewCatalogTests<HiveCatalog> {
+
+  private HiveMetastoreSetup hiveMetastoreSetup;
+
+  @BeforeEach
+  public void before() throws Exception {
+    hiveMetastoreSetup = new HiveMetastoreSetup(Collections.emptyMap());
+  }
+
+  @AfterEach
+  public void after() throws Exception {
+    hiveMetastoreSetup.stopMetastore();
+  }
+
+  @Override
+  protected HiveCatalog catalog() {
+    return hiveMetastoreSetup.catalog;
+  }
+
+  @Override
+  protected Catalog tableCatalog() {
+    return hiveMetastoreSetup.catalog;
+  }
+
+  @Override
+  protected boolean requiresNamespaceCreate() {
+    return true;
+  }
+
+  // Override few tests which are using AlreadyExistsException instead of NoSuchViewException
```

Comment:
I have some of this problem for Nessie catalog extending `ViewCatalogTests`. Should we change ViewCatalogTests to accept multiple options like I did for Nessie (https://github.com/apache/iceberg/pull/8909/files#diff-9c8b69f86aec86aea82cd6675c1c24267d9a9d77d7715c2c792934cb954b328bR403-R409) or generalize the code to throw expected exception?